space-framework/docs/space-framework.md

```diff
@@ -0,0 +1,92 @@
+
+**Space Object**, **Space** - A kind of Kubernetes API object that _represents_ a pSpace.  Its "kind" is `Space`, we sometimes include the "object" part when writing specifically about the API objects to clearly distinguish them from what they represent.
+
+**Space Provider** - A Thing that manages the life cycle of multiple pSpaces. This includes at minimal create/delete/List operations on pSpaces. KIND is an example for Space Provider (that manages KIND clusters), KCP is another example - KCP manages KCP's workspaces. 
+
+## High level architecture  & Main componenets
+![SF overall view](SF-All.drawio.svg "SF overall viewgit s")
+
+The Space Framework (SF) is a generic management framework for space providers and pSpaces. The framework defines an abstraction layer for space providers and pSpace management that allows clients (both CLI/script based and client-go based clients) to use spaces while mantaining the clients decoupled from the specific pSpace and space provider that is being used. 
+
+The The space framework uses two CRDs:
+- **SpaceProviderDesc:** This CRD represnts a space provider it holds the information needed in order to interact with a specific Space  provider. 
+- **Space:** This CRD represnts a space as defined above. It includes reference to secrets that allow clients to connect to the pSpace that it represents and also several attributes used by the SF that will be described later on. 
+
+The space fromework supports several ways to add new spaces to the frameowrk, this is controlled by definiing differet space types: 
+
+1. Managed Space: The pSpace is created through an explicit request to the management api server to create a new Space. For Managed spaces the space framework initiates a request to the Space provider to create the pSpace, the space frame work will then continously reconsile between the Space and the pSpace states. For managed spaces the desired state is controled through the Space object. For example, in order to delete the pSpace the client need to delete the corresponding Space object and the SF will interact with the space provider in order to delete the pSpace.
+2. Imported spaces: A Space can be created by importing an existing pSpace. In this case the Space is not linked to any space provider and the client who imports that Space need to supply the access information that allows other clients to connect to the pSpace. For imported spaces the desired state is derived by the pSpace. This means that the SF is not responsible for the life cycle management of the pSpace but only update the corresponding Space object. For example, when deleteing the space object of an existing imported pSpace the space framework will recreate the space object. 
```

Comment:
> when deleteing the space object of an existing imported pSpace the space framework will recreate the space object

Where is the information about this space stored besides the Space object?

Reply:
Secrets that the Space object reference

Reply:
continued in a new comment